lib/repositories/message.js

```diff
@@ -394,20 +412,89 @@ function createApi(table, provider, sqlGenerator, dependencies) {
                 });
             });
         });
+    },
+
+    /**
+     * Deletes all messages belonging to the specific mailbox
+     */
+    removeByMailbox: function remove(mailbox) {
+      dependencies.logger.trace('message.removeByMailbox called');
+
+      var query = table
+        .select(table.star())
+        .from(table)
+        .where(table['mailbox_id'].equals(mailbox.getId()))
+        .toQuery();
+      query = provider.forUpdate(query);
+
+      return provider.beginTransaction(true)
+        .then(function(transaction) {
+
+          return transaction.runQuery(query)
+            .then(function(result) {
+              var query = table
+                .delete()
+                .where(table['mailbox_id'].equals(mailbox.getId()))
+                .toQuery();
+
+              var messages = result.rows;
+
+              var x;
+              for (x in messages) {
```

Comment:
I would duplicate how this is done in other parts of the DAL and use result.reduce to construct the output.